vignettes/create-run-population.Rmd

```diff
@@ -52,7 +52,8 @@ populationCharacteristics <- createPopulationCharacteristics(
 print(populationCharacteristics)
 
 # Create population from population characteristics
-myPopulation <- createPopulation(populationCharacteristics = populationCharacteristics)
+result <- createPopulation(populationCharacteristics = populationCharacteristics)
+myPopulation <- result$population
```

Comment:
All: This is the change in API. Instead of getting a population object directly, one needs to get the $population object from the result.

Reply:
Why not just extend the population R class with the `DerivedParametersPaths `and `DerivedParametersValues `instead?

Reply:
because this would mean changing everything in core. AT the moment, a population is what is being varied. The concept of derived etc.. does not exist. It is also possible to override a formula with some values so this would create a serious headache. I don't think this is something we would want to do

Reply:
I disagree.
You could just add `derivedParameters` property to the **population R6-class only** (the property would be **read/write for now**. As soon as the core is adjusted, it can be made read-only). 
So instead of `return(list(population = population, derivedParameters = derivedParameters))` I would just do
```
population.derivedParameters = derivedParameters
return population
```
Then no API change now (and no API change in the future when the core is adjusted).

Reply:
We are going to have to agree to disagree. While this would work, for the population object, this is also the object that is being passed to runSimulation. 
Or do you suggest to create a separate list in the population object?

Reply:
>population.derivedParameters = derivedParameters

I see. The problem with this is that we have to add much more code. For example, we need to verify that a parameter is not in the two lists, etc. 

For simple consistency, because createIndividual returns a list with two items, I think createPopulation should do the same

Reply:
Let's discuss it tomorrow. I don't undersnad what readonly means for example. one should be allowed to overwrite this stuff.
So let's think about what we would like to have from an API point of view and we can update

Reply:
> > population.derivedParameters = derivedParameters
> 
> I see. The problem with this is that we have to add much more code. For example, we need to verify that a parameter is not in the two lists, etc.
> 
> For simple consistency, because createIndividual returns a list with two items, I think createPopulation should do the same

But `createIndividual` returns two lists of parameters. `createPopulation` would return a population-object and a list of parameters - rather inconvenient for me.

Reply:
Why is it iconvenenient? Those derived parmaeters are for "info" only and are stale as soon as you work with the population parameters (in a way that would change those derived parameters). I think that this is the best way to not have inconsistencies